File: src/Application Specific/Classes/Entity Collections/ContractOriginalSiteRates/ContractOriginalSiteRatesSQL.vb
Imports System.Data.SqlClient

Namespace Entity

Namespace ContractOriginalSiteRatess

Public Class ContractOriginalSiteRatesSQL
    Private _database As Sys.Database

    Public Sub New(ByVal database As Sys.Database)
        _database = database
    End Sub

#Region "Functions"

	
    Public Sub Delete(ByVal ContractOriginalSiteRateID As Integer)
    		_database.ClearParameter()
            _database.AddParameter("@ContractOriginalSiteRateID", ContractOriginalSiteRateID, True)
            _database.ExecuteSP_NO_Return("ContractOriginalSiteRates_Delete")
    End Sub

    Public Function [ContractOriginalSiteRates_Get](ByVal ContractOriginalSiteRateID As Integer) As ContractOriginalSiteRates
        _database.ClearParameter()
        _database.AddParameter("@ContractOriginalSiteRateID", ContractOriginalSiteRateID)
        
        'Get the datatable from the database store in dt
        Dim dt As DataTable = _database.ExecuteSP_DataTable("ContractOriginalSiteRates_Get")
        
        If Not dt Is Nothing Then			
			if dt.rows.count > 0 then
				Dim oContractOriginalSiteRates As New ContractOriginalSiteRates
				With oContractOriginalSiteRates
				    .IgnoreExceptionsOnSetMethods = True
						.SetContractOriginalSiteRateID= dt.Rows(0).Item("ContractOriginalSiteRateID")
				.SetContractSiteID= dt.Rows(0).Item("ContractSiteID")
				.SetRateID= dt.Rows(0).Item("RateID")
				.SetQty= dt.Rows(0).Item("Qty")

					.SetDeleted = dt.Rows(0).Item("Deleted")               
				End With		
				oContractOriginalSiteRates.Exists = true				
			Return oContractOriginalSiteRates
			else
					Return Nothing
			end if 
			       Else
            Return Nothing
        End If
            End Function

            Public Function ContractOriginalSiteRates_GetForContractSite(ByVal ContractSiteID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@ContractSiteID", ContractSiteID)

                Dim dt As DataTable = _database.ExecuteSP_DataTable("ContractOriginalSiteRates_GetForContractSite")
                dt.TableName = Entity.Sys.Enums.TableNames.tblContractOriginalSiteRates.ToString
                Return New DataView(dt)

            End Function

    Public Function [ContractOriginalSiteRates_GetAll]() As DataView
       _database.ClearParameter()
        Dim dt As DataTable = _database.ExecuteSP_DataTable("ContractOriginalSiteRates_GetAll")
        dt.TableName = Entity.Sys.Enums.TableNames.tblContractOriginalSiteRates.ToString
        Return New DataView(dt)
    End Function
   
    Public Function [Insert](ByVal oContractOriginalSiteRates As ContractOriginalSiteRates) As ContractOriginalSiteRates
		 _database.ClearParameter()			
	     AddContractOriginalSiteRatesParametersToCommand(oContractOriginalSiteRates)	
						
		oContractOriginalSiteRates.SetContractOriginalSiteRateID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("ContractOriginalSiteRates_Insert"))				
        oContractOriginalSiteRates.Exists = True
        Return oContractOriginalSiteRates
    End Function   
    

    Public Sub [Update](ByVal oContractOriginalSiteRates As ContractOriginalSiteRates)
		_database.ClearParameter()
		_database.AddParameter("@ContractOriginalSiteRateID", oContractOriginalSiteRates.ContractOriginalSiteRateID, True)  
		AddContractOriginalSiteRatesParametersToCommand(oContractOriginalSiteRates)
		_database.ExecuteSP_NO_Return("ContractOriginalSiteRates_Update")
    End Sub



    Private Sub AddContractOriginalSiteRatesParametersToCommand(ByRef oContractOriginalSiteRates As ContractOriginalSiteRates)    
        With _database
			.AddParameter("@ContractSiteID", oContractOriginalSiteRates.ContractSiteID, True)
			.AddParameter("@RateID", oContractOriginalSiteRates.RateID, True)
			.AddParameter("@Qty", oContractOriginalSiteRates.Qty, True)

		
            
        End With        
    End Sub      
    

#End Region

End Class

End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/ContractOriginalSiteRates/ContractOriginalSiteRatesSQL.cs
using System.Data;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace ContractOriginalSiteRatess
    {
        public class ContractOriginalSiteRatesSQL
        {
            private Sys.Database _database;

            public ContractOriginalSiteRatesSQL(Sys.Database database)
            {
                _database = database;
            }

            public void Delete(int ContractOriginalSiteRateID)
            {
                _database.ClearParameter();
                _database.AddParameter("@ContractOriginalSiteRateID", ContractOriginalSiteRateID, true);
                _database.ExecuteSP_NO_Return("ContractOriginalSiteRates_Delete");
            }

            public DataView ContractOriginalSiteRates_GetForContractSite(int ContractSiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@ContractSiteID", ContractSiteID);
                var dt = _database.ExecuteSP_DataTable("ContractOriginalSiteRates_GetForContractSite");
                dt.TableName = Sys.Enums.TableNames.tblContractOriginalSiteRates.ToString();
                return new DataView(dt);
            }

            public ContractOriginalSiteRates Insert(ContractOriginalSiteRates oContractOriginalSiteRates)
            {
                _database.ClearParameter();
                AddContractOriginalSiteRatesParametersToCommand(ref oContractOriginalSiteRates);
                oContractOriginalSiteRates.SetContractOriginalSiteRateID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("ContractOriginalSiteRates_Insert"));
                oContractOriginalSiteRates.Exists = true;
                return oContractOriginalSiteRates;
            }

            public void Update(ContractOriginalSiteRates oContractOriginalSiteRates)
            {
                _database.ClearParameter();
                _database.AddParameter("@ContractOriginalSiteRateID", oContractOriginalSiteRates.ContractOriginalSiteRateID, true);
                AddContractOriginalSiteRatesParametersToCommand(ref oContractOriginalSiteRates);
                _database.ExecuteSP_NO_Return("ContractOriginalSiteRates_Update");
            }

            private void AddContractOriginalSiteRatesParametersToCommand(ref ContractOriginalSiteRates oContractOriginalSiteRates)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@ContractSiteID", oContractOriginalSiteRates.ContractSiteID, true);
                    withBlock.AddParameter("@RateID", oContractOriginalSiteRates.RateID, true);
                    withBlock.AddParameter("@Qty", oContractOriginalSiteRates.Qty, true);
                }
            }
        }
    }
}